### PR TITLE
Remove v1-related code from medusa app

### PR DESCRIPTION
**Pull request by sradevski**

### CODE REVIEW

File: packages/medusa/package.json [hunk 1]
```diff
@@ -63,46 +61,36 @@
...
     "pg": "^8.11.2",
     "qs": "^6.11.2",
-    "randomatic": "^3.1.1",
     "reflect-metadata": "^0.1.13",
     "regenerator-runtime": "^0.13.11",
     "request-ip": "^3.3.0",
-    "scrypt-kdf": "^2.0.1",
-    "ulid": "^2.3.0",
     "uuid": "^9.0.0",
-    "winston": "^3.8.2",
```
Comment (adrien2p) [inline packages/medusa/package.json:105]: Q: aren't we using the logger anymore?

File: packages/medusa/src/api-v2/admin/product-categories/validators.ts [hunk 2]
```diff
@@ -35,22 +29,10 @@ export const AdminProductCategoriesParams = createFindParams({
...
-    ),
-    is_internal: z.preprocess(
-      (val: any) => optionalBooleanMapper.get(val?.toLowerCase()),
-      z.boolean().optional()
-    ),
-    is_active: z.preprocess(
-      (val: any) => optionalBooleanMapper.get(val?.toLowerCase()),
-      z.boolean().optional()
-    ),
+    include_ancestors_tree: OptionalBooleanValidator,
```
Comment (adrien2p) [inline packages/medusa/src/api-v2/admin/product-categories/validators.ts:32]: 🙏 nice refactoring

File: packages/medusa/src/commands/utils/get-migrations.js [hunk 3]
```diff
@@ -1,98 +1,6 @@
...
-      name: packageJSON.name,
-      version: packageJSON.version,
-    }
-  } catch (err) {
-    throw new Error(
-      `Unable to find plugin "${pluginName}". Perhaps you need to install its package?`
-    )
-  }
-}
+import { isDefined } from "medusa-core-utils"
```
Comment (adrien2p) [inline packages/medusa/src/commands/utils/get-migrations.js:3]: todo: from the utils package

**Review by adrien2p — APPROVED** [review]
LGTM, missing the changes in the modules sdk definition for the dependencies around the event bus service I believe

File: packages/medusa/package.json (hunk 1 above)
Comment (sradevski) [reply packages/medusa/package.json:105]: @adrien2p winston wasn't used anywhere in the codebase it seems

File: packages/medusa/src/commands/utils/get-migrations.js (hunk 3 above)
Comment (sradevski) [reply packages/medusa/src/commands/utils/get-migrations.js:3]: @adrien2p there are still some small cleanups left, I will go through everything one more time and clean up any orphan code, it was just difficult to do it all in 1 go and have it reviewable 😄

Comment (adrien2p) [reply packages/medusa/src/commands/utils/get-migrations.js:3]: just doubled check and yes you are right :D

File: packages/medusa/package.json [hunk 4]
```diff
@@ -38,7 +38,6 @@
     "rimraf": "^5.0.1",
     "supertest": "^4.0.2",
     "ts-jest": "^25.5.1",
-    "typeorm": "^0.3.16",
```
Comment (riqwan) [inline packages/medusa/package.json:41]: this has to be my favorite line change :D 

cc @adrien2p

File: packages/medusa/src/loaders/helpers/routing/__fixtures__/server/index.js [hunk 5]
```diff
@@ -88,10 +84,6 @@ export const createServer = async (rootDir) => {
   })
 
   featureFlagLoader(config)
-  modelsLoader({ container, isTest: true })
-  repositoriesLoader({ container, isTest: true })
-  servicesLoader({ container, configModule: config })
-  strategiesLoader({ container, configModule: config })
   await passportLoader({ app: app, container, configModule: config })
```
Comment (riqwan) [inline packages/medusa/src/loaders/helpers/routing/__fixtures__/server/index.js:87]: do we still use passport? I think we can can this as well

**Review by riqwan — APPROVED** [review]
🔥 😮‍💨

File: packages/medusa/src/loaders/helpers/routing/__fixtures__/server/index.js (hunk 5 above)
Comment (sradevski) [reply packages/medusa/src/loaders/helpers/routing/__fixtures__/server/index.js:87]: @riqwan I left some things I wasn't 100% sure of, but I think we don't. I will do a follow-up PR with the smaller cleanups, wanted to get the bulk out of the way.

**Pull request merged**